CMakeLists.txt

```diff
@@ -69,7 +69,21 @@ FIND_PACKAGE(deal.II 8.4.0 REQUIRED
   HINTS ${deal.II_DIR} ${DEAL_II_DIR} ../ ../../ $ENV{DEAL_II_DIR}
   )
 
+# If the CMAKE_BUILD_TYPE is set via command line we keep this one,
+# otherwise it is set equal to the one of deal.II library
+
+IF(DEFINED CMAKE_BUILD_TYPE)
+   SET(CMAKE_BUILD_TYPE ${CMAKE_BUILD_TYPE} CACHE STRING 
+       "Choose the type of build, options are: Debug, Release or DebugRelease"
+       )
+ELSE()
+   SET(CMAKE_BUILD_TYPE ${DEAL_II_BUILD_TYPE} CACHE STRING
+       "Choose the type of build, options are: Debug, Release or DebugRelease"
+       )
+ENDIF()
```

Comment:
There is no need for the `IF`-`ELSE` block here. You can simply set unconditionally
```
SET(CMAKE_BUILD_TYPE ${DEAL_II_BUILD_TYPE} CACHE STRING 
  "Choose the type of build, options are: Debug, Release or DebugRelease"
  )
```

If the variable `CMAKE_BUILD_TYPE` is already set in cache, the `SET(... CACHE ...)` function will not override it.
(If no additional `FORCE` is supplied as keyword.)